static/js/components/dashboard/FinalExamCard.js

```diff
@@ -175,25 +200,44 @@ export default class FinalExamCard extends React.Component<void, Props, void> {
       navigateToProfile,
       submitPearsonSSO,
       pearson,
+      ui: { dialogVisibility: { showPearsonTosDialog = false } },
+      showToPearsonSiteDialog
     } = this.props;
 
     if (!SETTINGS.FEATURES.EXAMS) {
       return null;
     }
 
+    let content = null;
     switch (program.pearson_exam_status) {
     case PEARSON_PROFILE_ABSENT:
-      return absentCard();
+      content = absentCard();
+      break;
     case PEARSON_PROFILE_SUCCESS:
-      return successCard(profile, navigateToProfile);
+      content = successCard(profile, navigateToProfile);
+      break;
     case PEARSON_PROFILE_IN_PROGRESS:
-      return pendingCard();
+      content = pendingCard();
+      break;
     case PEARSON_PROFILE_INVALID:
-      return invalidCard(navigateToProfile);
+      content = invalidCard(navigateToProfile);
+      break;
     case PEARSON_PROFILE_SCHEDULABLE:
-      return schedulableCard(profile, program, navigateToProfile, submitPearsonSSO, pearson);
-    default:
-      return null;
+      content = schedulableCard(
+        profile,
+        program,
+        navigateToProfile,
+        pearson,
+        showToPearsonSiteDialog
+      );
+      break;
     }
+
+    return (
+      <div>
+        {confirmPearsonTOSDialog(showPearsonTosDialog, showToPearsonSiteDialog, submitPearsonSSO)}
```

Comment:
somehow it is not working when i put or pass this dialog in `schedulableCard`. it only works when i put it at root